### PR TITLE
Add JS/WASM target

### PR DESCRIPTION
**Pull request by zacharee**
This change adds a target for browsers.

### CODE REVIEW

File: .idea/workspace.xml [hunk 1]
```diff
@@ -1 +0,0 @@
-<project version="4"><component name="CodeInsightSettings"><option name="OPTIMIZE_IMPORTS_ON_THE_FLY" value="true"/></component></project>
```
Comment (zacharee) [inline .idea/workspace.xml:1]: This is in the `.gitignore` so I removed it.

File: precompose/src/commonMain/kotlin/moe/tlaster/precompose/ui/ViewModelAdapter.kt [hunk 2]
```diff
@@ -33,7 +33,7 @@ private fun <T : ViewModel> ViewModelStoreOwner.getViewModel(
     modelClass: KClass<T>,
     creator: () -> T,
 ): T {
-    val key = (keys.map { it.hashCode().toString() } + modelClass.qualifiedName).joinToString()
+    val key = (keys.map { it.hashCode().toString() } + modelClass.simpleName).joinToString()
```
Comment (zacharee) [inline precompose/src/commonMain/kotlin/moe/tlaster/precompose/ui/ViewModelAdapter.kt:36]: KotlinJS doesn't support `qualifiedName` yet.

File: precompose/src/commonMain/kotlin/moe/tlaster/precompose/viewmodel/ViewModelProvider.kt [hunk 3]
```diff
@@ -5,7 +5,7 @@ import kotlin.reflect.KClass
 inline fun <reified T : ViewModel> ViewModelStore.getViewModel(
     noinline creator: () -> T,
 ): T {
-    val key = T::class.qualifiedName.toString()
+    val key = T::class.simpleName.toString()
```
Comment (zacharee) [inline precompose/src/commonMain/kotlin/moe/tlaster/precompose/viewmodel/ViewModelProvider.kt:8]: KotlinJS doesn't support `qualifiedName` yet.

File: precompose/src/jsMain/kotlin/moe/tlaster/precompose/PreComposeWindow.kt [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        setContent {
+            val holder = remember {
+                PreComposeWindowHolder()
+            }
+            ProvideJsCompositionLocals(holder) {
+                content.invoke(this)
+            }
+        }
+    }
+}
```
Comment (zacharee) [inline precompose/src/jsMain/kotlin/moe/tlaster/precompose/PreComposeWindow.kt:38]: The WASM version of Skiko doesn't handle window resizes yet, so callers have to have access to the `ComposeWindow` instance in order to handle resizes manually. This should only be needed temporarily.

File: sample/js/src/jsMain/kotlin/main.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+    val scale = layer.layer.contentScale
+    canvasResize()
+    layer.layer.attachTo(clone)
+    layer.layer.needRedraw()
+    layer.setSize(
+        (clone.width / scale).toInt(),
+        (clone.height / scale).toInt()
+    )
+}
```
Comment (zacharee) [inline sample/js/src/jsMain/kotlin/main.kt:34]: This is the code for manually handling window resizes in Skiko WASM.

**Review by Tlaster — APPROVED** [review]
LGTM!

**Pull request merged**